arangod/Cluster/AgencyCallback.cpp

```diff
@@ -203,7 +202,11 @@ bool AgencyCallback::executeByCallbackOrTimeout(double maxTimeout) {
 
     // we haven't yet been signaled. so let's wait for a signal or
     // the timeout to occur
-    if (!_cv.wait(static_cast<uint64_t>(maxTimeout * 1000000.0))) {
+    std::unique_lock lock{_cv.mutex};
+    if (_cv.cv.wait_for(lock, std::chrono::microseconds{static_cast<uint64_t>(
+                                  maxTimeout * 1000000.0)}) ==
+        std::cv_status::timeout) {
+      lock.unlock();
```

Comment:
Incorrect replacement, looks like _cv.mutex already was locked here.

So needs to write
```suggestion
    std::unique_lock lock{_cv.mutex, std::adopt_lock};
    auto const cv_status = _cv.cv.wait_for(lock, std::chrono::microseconds{static_cast<uint64_t>(
                                  maxTimeout * 1000000.0)});
    lock.release();
    if (cv_status == std::cv_status::timeout) {
```


Btw it's place, where in old code tsan detected lock order inversion.
Maybe someone should fix it?

Reply:
Suggestion applied